<?xml version="1.0"?>
<robot name="ur10" version="1.0">
  <transmission name="shoulder_pan_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_pan_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_pan_motor">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="shoulder_lift_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_lift_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_lift_motor">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="elbow_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="elbow_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="elbow_motor">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_1_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_1_motor">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_2_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_2_motor">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="wrist_3_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_3_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_3_motor">
      <mechanicalReduction>1.0</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo>
    <plugin filename="libgazebo_ros2_control.so" name="gazebo_ros2_control">
      <robot_sim_type>gazebo_ros2_control/GazeboSystem</robot_sim_type>
      <parameters>/home/rshailly/dev_ws/install/ur10_ros2_gazebo/share/ur10_ros2_gazebo/config/ur10_controller.yaml</parameters>
    </plugin>
  </gazebo>
  <gazebo reference="shoulder_link">
    <selfCollide>false</selfCollide>
  </gazebo>
  <gazebo reference="upper_arm_link">
    <selfCollide>false</selfCollide>
  </gazebo>
  <gazebo reference="forearm_link">
    <selfCollide>false</selfCollide>
  </gazebo>
  <gazebo reference="wrist_1_link">
    <selfCollide>false</selfCollide>
  </gazebo>
  <gazebo reference="wrist_3_link">
    <selfCollide>false</selfCollide>
  </gazebo>
  <gazebo reference="wrist_2_link">
    <selfCollide>false</selfCollide>
  </gazebo>
  <gazebo reference="ee_link">
    <selfCollide>false</selfCollide>
  </gazebo>
  <link name="base_link"/>
  <link name="base_link_inertia">
    <inertial>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
      <mass value="4.0"/>
      <inertia ixx="0.0061063308908" ixy="0.0" ixz="0.0" iyy="0.0061063308908" iyz="0.0" izz="0.01125"/>
    </inertial>
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 3.141592653589793"/>
      <geometry>
        <mesh filename="file:///home/rshailly/dev_ws/install/ur10_ros2_gazebo/share/ur10_ros2_gazebo/meshes/visual/base.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 3.141592653589793"/>
      <geometry>
        <mesh filename="file:///home/rshailly/dev_ws/install/ur10_ros2_gazebo/share/ur10_ros2_gazebo/meshes/collision/base.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="shoulder_link">
    <inertial>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
      <mass value="7.778"/>
      <inertia ixx="0.03147431257693659" ixy="0.0" ixz="0.0" iyy="0.03147431257693659" iyz="0.0" izz="0.021875624999999996"/>
    </inertial>
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 3.141592653589793"/>
      <geometry>
        <mesh filename="file:///home/rshailly/dev_ws/install/ur10_ros2_gazebo/share/ur10_ros2_gazebo/meshes/visual/shoulder.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 3.141592653589793"/>
      <geometry>
        <mesh filename="file:///home/rshailly/dev_ws/install/ur10_ros2_gazebo/share/ur10_ros2_gazebo/meshes/collision/shoulder.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="upper_arm_link">
    <inertial>
      <origin xyz="-0.306 0.0 0.175" rpy="0.0 1.5707963267948966 0.0"/>
      <mass value="12.93"/>
      <inertia ixx="0.42175380379841093" ixy="0.0" ixz="0.0" iyy="0.42175380379841093" iyz="0.0" izz="0.03636562499999999"/>
    </inertial>
    <visual>
      <origin xyz="0.0 0.0 0.220941" rpy="1.5707963267948966 0.0 -1.5707963267948966"/>
      <geometry>
        <mesh filename="file:///home/rshailly/dev_ws/install/ur10_ros2_gazebo/share/ur10_ros2_gazebo/meshes/visual/upperarm.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0.0 0.0 0.220941" rpy="1.5707963267948966 0.0 -1.5707963267948966"/>
      <geometry>
        <mesh filename="file:///home/rshailly/dev_ws/install/ur10_ros2_gazebo/share/ur10_ros2_gazebo/meshes/collision/upperarm.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="forearm_link">
    <inertial>
      <origin xyz="-0.28615 0.0 0.049042" rpy="0.0 1.5707963267948966 0.0"/>
      <mass value="3.87"/>
      <inertia ixx="0.11106969409710458" ixy="0.0" ixz="0.0" iyy="0.11106969409710458" iyz="0.0" izz="0.010884375"/>
    </inertial>
    <visual>
      <origin xyz="0.0 0.0 0.049042" rpy="1.5707963267948966 0.0 -1.5707963267948966"/>
      <geometry>
        <mesh filename="file:///home/rshailly/dev_ws/install/ur10_ros2_gazebo/share/ur10_ros2_gazebo/meshes/visual/forearm.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0.0 0.0 0.049042" rpy="1.5707963267948966 0.0 -1.5707963267948966"/>
      <geometry>
        <mesh filename="file:///home/rshailly/dev_ws/install/ur10_ros2_gazebo/share/ur10_ros2_gazebo/meshes/collision/forearm.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="wrist_1_link">
    <inertial>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
      <mass value="1.96"/>
      <inertia ixx="0.005108247956699999" ixy="0.0" ixz="0.0" iyy="0.005108247956699999" iyz="0.0" izz="0.005512499999999999"/>
    </inertial>
    <visual>
      <origin xyz="0.0 0.0 -0.1149" rpy="1.5707963267948966 0.0 0.0"/>
      <geometry>
        <mesh filename="file:///home/rshailly/dev_ws/install/ur10_ros2_gazebo/share/ur10_ros2_gazebo/meshes/visual/wrist1.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0.0 0.0 -0.1149" rpy="1.5707963267948966 0.0 0.0"/>
      <geometry>
        <mesh filename="file:///home/rshailly/dev_ws/install/ur10_ros2_gazebo/share/ur10_ros2_gazebo/meshes/collision/wrist1.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="wrist_2_link">
    <inertial>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
      <mass value="1.96"/>
      <inertia ixx="0.0040792483683" ixy="0.0" ixz="0.0" iyy="0.0040792483683" iyz="0.0" izz="0.005512499999999999"/>
    </inertial>
    <visual>
      <origin xyz="0.0 0.0 -0.1158" rpy="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="file:///home/rshailly/dev_ws/install/ur10_ros2_gazebo/share/ur10_ros2_gazebo/meshes/visual/wrist2.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0.0 0.0 -0.1158" rpy="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="file:///home/rshailly/dev_ws/install/ur10_ros2_gazebo/share/ur10_ros2_gazebo/meshes/collision/wrist2.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="wrist_3_link">
    <inertial>
      <origin xyz="0.0 0.0 -0.01525" rpy="0.0 0.0 0.0"/>
      <mass value="0.202"/>
      <inertia ixx="0.00011792166116465" ixy="0.0" ixz="0.0" iyy="0.00011792166116465" iyz="0.0" izz="0.00020452500000000002"/>
    </inertial>
    <visual>
      <origin xyz="0.0 0.0 -0.0922" rpy="1.5707963267948966 0.0 0.0"/>
      <geometry>
        <mesh filename="file:///home/rshailly/dev_ws/install/ur10_ros2_gazebo/share/ur10_ros2_gazebo/meshes/visual/wrist3.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <origin xyz="0.0 0.0 -0.0922" rpy="1.5707963267948966 0.0 0.0"/>
      <geometry>
        <mesh filename="file:///home/rshailly/dev_ws/install/ur10_ros2_gazebo/share/ur10_ros2_gazebo/meshes/collision/wrist3.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="base_link-base_link_inertia" type="fixed">
    <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 3.141592653589793"/>
    <parent link="base_link"/>
    <child link="base_link_inertia"/>
  </joint>
  <joint name="shoulder_pan_joint" type="revolute">
    <origin xyz="0.0 0.0 0.1273" rpy="0.0 0.0 0.0"/>
    <axis xyz="0.0 0.0 1.0"/>
    <parent link="base_link_inertia"/>
    <child link="shoulder_link"/>
    <limit effort="330.0" lower="-6.2832" upper="6.2832" velocity="2.0944"/>
    <dynamics damping="50.0" friction="1.0"/>
  </joint>
  <joint name="shoulder_lift_joint" type="revolute">
    <origin xyz="0.0 0.0 0.0" rpy="1.570796327 0.0 0.0"/>
    <axis xyz="0.0 0.0 1.0"/>
    <parent link="shoulder_link"/>
    <child link="upper_arm_link"/>
    <limit effort="330.0" lower="-6.2832" upper="6.2832" velocity="2.0944"/>
    <dynamics damping="50.0" friction="1.0"/>
  </joint>
  <joint name="elbow_joint" type="revolute">
    <origin xyz="-0.612 0.0 0.0" rpy="0.0 0.0 0.0"/>
    <axis xyz="0.0 0.0 1.0"/>
    <parent link="upper_arm_link"/>
    <child link="forearm_link"/>
    <limit effort="150.0" lower="-3.1416" upper="3.1416" velocity="3.1416"/>
    <dynamics damping="10.0" friction="1.0"/>
  </joint>
  <joint name="wrist_1_joint" type="revolute">
    <origin xyz="-0.5723 0.0 0.163941" rpy="0.0 0.0 0.0"/>
    <axis xyz="0.0 0.0 1.0"/>
    <parent link="forearm_link"/>
    <child link="wrist_1_link"/>
    <limit effort="56.0" lower="-6.2832" upper="6.2832" velocity="3.1416"/>
    <dynamics damping="5.0" friction="1.0"/>
  </joint>
  <joint name="wrist_2_joint" type="revolute">
    <origin xyz="0.0 -0.1157 -2.373046667922381e-11" rpy="1.570796327 0.0 0.0"/>
    <axis xyz="0.0 0.0 1.0"/>
    <parent link="wrist_1_link"/>
    <child link="wrist_2_link"/>
    <limit effort="56.0" lower="-6.2832" upper="6.2832" velocity="3.1416"/>
    <dynamics damping="2.0" friction="1.0"/>
  </joint>
  <joint name="wrist_3_joint" type="revolute">
    <origin xyz="0.0 0.0922 -1.891053610911353e-11" rpy="1.570796326589793 3.141592653589793 3.141592653589793"/>
    <axis xyz="0.0 0.0 1.0"/>
    <parent link="wrist_2_link"/>
    <child link="wrist_3_link"/>
    <limit effort="56.0" lower="-6.2832" upper="6.2832" velocity="3.1416"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="base"/>
  <joint name="base_link-base_fixed_joint" type="fixed">
    <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 3.141592653589793"/>
    <parent link="base_link"/>
    <child link="base"/>
  </joint>
  <link name="flange"/>
  <joint name="wrist_3-flange" type="fixed">
    <origin xyz="0.0 0.0 0.0" rpy="0.0 -1.5707963267948966 -1.5707963267948966"/>
    <parent link="wrist_3_link"/>
    <child link="flange"/>
  </joint>
  <link name="tool0"/>
  <joint name="flange-tool0" type="fixed">
    <origin xyz="0.0 0.0 0.0" rpy="1.5707963267948966 0.0 1.5707963267948966"/>
    <parent link="flange"/>
    <child link="tool0"/>
  </joint>
  <link name="world"/>
  <link name="ur_base">
    <inertial>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
      <mass value="100.0"/>
      <inertia ixx="0.01" ixy="0.0" ixz="0.0" iyy="0.01" iyz="0.0" izz="0.01"/>
    </inertial>
    <collision>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
      <geometry>
        <box size="0.2 0.2 1.3"/>
      </geometry>
    </collision>
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
      <geometry>
        <box size="0.2 0.2 1.3"/>
      </geometry>
      <material name="red">
        <color rgba="1.0 0.0 0.0 1.0"/>
      </material>
    </visual>
  </link>
  <gazebo reference="ur_base">
    <material>Gazebo/Red</material>
  </gazebo>
  <joint name="world_joint" type="fixed">
    <origin xyz="0.0 0.0 0.65" rpy="0.0 0.0 0.0"/>
    <parent link="world"/>
    <child link="ur_base"/>
  </joint>
  <joint name="ur_joint" type="fixed">
    <origin xyz="0.0 0.0 0.655" rpy="0.0 0.0 0.0"/>
    <parent link="ur_base"/>
    <child link="base_link"/>
  </joint>
</robot>
